src/registryDAO.mligo

```diff
@@ -54,34 +54,49 @@ let apply_diff_affected
   let max_proposal_size = unpack_nat(find_big_map("max_proposal_size", extras)) in
 
   let required_token_lock = frozen_scale_value * proposal_size + frozen_extra_value in
-  let has_correct_token_lock =
-    (params.frozen_token = required_token_lock) && (proposal_size < max_proposal_size) in
-
-  if has_correct_token_lock then
-    match unpack_proposal_metadata(params.proposal_metadata) with
-    | Transfer_proposal tp ->
-        let min_xtz_amount = unpack_tez(find_big_map("min_xtz_amount", extras)) in
-        let max_xtz_amount = unpack_tez(find_big_map("max_xtz_amount", extras)) in
-        let is_all_transfers_valid (is_valid, transfer_type: bool * transfer_type) =
-          match transfer_type with
-          | Token_transfer_type _tt -> is_valid
-          | Xtz_transfer_type xt ->
-                 is_valid && xt.amount <> 0mutez
-              && min_xtz_amount <= xt.amount && xt.amount <= max_xtz_amount
-        in
-        List.fold is_all_transfers_valid tp.transfers has_correct_token_lock
-    | Update_receivers_proposal _urp -> has_correct_token_lock
-    | Configuration_proposal _cp -> has_correct_token_lock
-
-  else
-    false
+  let pm = unpack_proposal_metadata(params.proposal_metadata) in
+
+  let _ : unit =
+    if (params.frozen_token <> required_token_lock) then
+      ([%Michelson ({| { FAILWITH } |} : (string * string) -> unit)]
+        ("FAIL_PROPOSAL_CHECK", "Incorrect token amounts locked") : unit)
+    else if (proposal_size >= max_proposal_size) then
+       ([%Michelson ({| { FAILWITH } |} : (string * string) -> unit)]
+        ("FAIL_PROPOSAL_CHECK", "Proposal size is bigger than 'max_proposal_size'") : unit)
+    else
+      unit in
+
+  match unpack_proposal_metadata(params.proposal_metadata) with
+  | Transfer_proposal tp ->
+      let min_xtz_amount = unpack_tez(find_big_map("min_xtz_amount", extras)) in
+      let max_xtz_amount = unpack_tez(find_big_map("max_xtz_amount", extras)) in
+      let is_all_transfers_valid (acc, transfer_type: unit * transfer_type) =
+        match transfer_type with
+        | Token_transfer_type _tt -> acc
+        | Xtz_transfer_type xt ->
+            if (xt.amount = 0mutez) then
+              ([%Michelson ({| { FAILWITH } |} : (string * string) -> unit)]
+                ("FAIL_PROPOSAL_CHECK", "Xtz transfer amount cannot be 0") : unit)
+            else if (xt.amount < min_xtz_amount) then
+              ([%Michelson ({| { FAILWITH } |} : (string * string) -> unit)]
+                ("FAIL_PROPOSAL_CHECK", "Xtz transfer amount cannot be smaller than 'min_xtz_amount'") : unit)
+            else if (xt.amount > max_xtz_amount) then
+              ([%Michelson ({| { FAILWITH } |} : (string * string) -> unit)]
+                ("FAIL_PROPOSAL_CHECK", "Xtz transfer amount cannot be bigger than 'max_xtz_amount'") : unit)
+            else
+              acc
+      in
+        List.fold is_all_transfers_valid tp.transfers unit
```

Comment:
I think we should now use `List.iter` instead

Reply:
Updated.